packages/plugin-global-user-properties/src/global-user-properties.ts

```diff
@@ -0,0 +1,25 @@
+import { EnrichmentPlugin, Event } from '@amplitude/analytics-types';
+import { GlobalUserPropertiesPlugin, Options } from './typings/global-user-properties';
+import { isAmplitudeIdentifyEvent, isNotSpecialAmplitudeEvent } from './helpers';
+
+export const globalUserPropertiesPlugin: GlobalUserPropertiesPlugin = function (options: Options = {}) {
+  const plugin: EnrichmentPlugin = {
+    name: '@amplitude/plugin-global-user-properties',
+    type: 'enrichment',
+
+    /* Note: The promise is because of the interface, not because this has any asynchronous behavior */
+    execute: async (event: Event): Promise<Event> => {
+      if (isNotSpecialAmplitudeEvent(event) || isAmplitudeIdentifyEvent(event)) {
```

Comment:
should we add a comment on why we don't attach globals to special events or is that well known logic already?

Reply:
I can add a comment - revenue charts do not have user properties!

Reply:
oh interesting! TIL!

Reply:
for other events which are not belong to amplitude special events, if there is no user_properties, do we want to set it to null?

Reply:
Please correct me if I'm wrong. 
Based on the definition of the `SpecialEventType`, it already includes the identify event. 

```
export enum SpecialEventType {
  IDENTIFY = '$identify',
  GROUP_IDENTIFY = '$groupidentify',
  REVENUE = 'revenue_amount',
}
```
`isNotSpecialAmplitudeEvent` means no event_type is '$identify', isAmplitudeIdentifyEvent means the event_type = '$identify'.
Does that conflict with each other?